openapi.yml

```diff
@@ -0,0 +1,475 @@
+      operationId: postWorkflow
+      requestBody:
+        description: 'Register a new workflow'
+        content:
+          'text/vnd.ocrd+sh': {}
+        required: true
+      responses:
+        '200':
+          description: Created a new OCR-D workflow
+          content: {application/json: {schema: {$ref: '#/components/schemas/Workflow'}}}
+
+  '/workflow/{workflow-id}':
+    put:
+      tags: ['workflow']
+      operationId: putWorkflow
+      parameters:
+        - name: workflow-id
+          in: path
+          description: ID of the Workflow
+          schema: {type: string}
+          required: true
+      requestBody:
+        description: 'Replace existing or create new workflow'
+        content:
+          'text/vnd.ocrd+sh': {}
+        required: true
+      responses:
+        '200':
+          description: Created a new OCR-D workflow
+          content: {application/json: {schema: {$ref: '#/components/schemas/Workflow'}}}
+    get:
+      tags: ['workflow', 'discovery']
+      operationId: getWorkflow
+      parameters:
+        - name: workflow-id
+          in: path
+          description: ID of the Workflow
+          schema: {type: string}
+          required: true
+      responses:
+        '200':
+          description: Return ProcessorJob
+          content:
+            application/json: {schema: {$ref: '#/components/schemas/Workflow'}}
+            application/vnd.ocrd+zip: {}
+        '404': {content: {}, description: 'Workflow not available'}
+    post:
+      tags: ['workflow']
+      operationId: runWorkflow
+      parameters:
+        - name: workflow-id
+          in: path
+          description: ID of the Workflow
+          schema: {type: string}
+          required: true
+      responses:
+        '200':
+          description: Return WorkflowJob
+          content:
+            application/json: {schema: {$ref: '#/components/schemas/WorkflowJob'}}
+
+  '/workflow/{workflow-id}/{job-id}':
+    get:
+      tags: ['workflow']
+      parameters:
+        - name: workflow-id
+          in: path
+          description: ID of the Workflow
+          schema: {type: string}
+          required: true
+        - name: job-id
+          in: path
+          description: ID of the ProcessorJob
+          schema: {type: string}
+          required: true
+      operationId: getWorkflowJob
+      responses:
+        '200':
+          description: Return WorkflowJob
+          content:
+            application/json: {schema: {$ref: '#/components/schemas/WorkflowJob'}}
+        '404':
+          description: WorkflowJob not found
+          content: {}
+
+  '/workspace':
+    get:
+      tags: ['workspace']
+      operationId: getWorkspaces
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Workspace'
+    post:
```

Comment:
Is there a scenario where a user would update a workspace (making a `PUT` necessary)?